rtp/src/rtcp/xr.rs

```diff
@@ -0,0 +1,258 @@
+use crate::{FeedbackMessageType, MediaTime, RtcpType, Ssrc};
+
+use super::{RtcpHeader, RtcpPacket};
+
+
+//   0                   1                   2                   3
+//   0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//   |V=2|P|reserved |   PT=XR=207   |             length            |
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//   |                              SSRC                             |
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//   :                         report blocks                         :
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+/// RFC 3611
+/// https://datatracker.ietf.org/doc/html/rfc3611#page-21
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ExtendedReport {
+    pub sender_ssrc: Ssrc,
+    pub blocks: Vec<ReportBlock>,
+}
+
+impl RtcpPacket for ExtendedReport {
+    fn header(&self) -> RtcpHeader {
+        RtcpHeader {
+            rtcp_type: RtcpType::ExtendedReport,
+            feedback_message_type: FeedbackMessageType::NotUsed,
+            words_less_one: (self.length_words() - 1) as u16,
+        }
+    }
+
+    fn length_words(&self) -> usize {
+        let header = 1;
+        let ssrc = 1;
+        let blocks: usize = self.blocks.iter().map(|b| b.len() / 4).sum();
+        header + ssrc + blocks
+    }
+
+    fn write_to(&self, buf: &mut [u8]) -> usize {
+        let mut len = self.header().write_to(buf);
+
+        buf[4..8].copy_from_slice(&self.sender_ssrc.to_be_bytes());
+        len += 4;
+
+        for block in self.blocks.iter() {
+            len += match block {
+                ReportBlock::ReceiverReferenceTime(b) => b.write_to(&mut buf[len..]),
+                ReportBlock::Dlrr(b) => b.write_to(&mut buf[len..]),
+            };
+        }
+
+        len
+    }
+}
+
+impl<'a> TryFrom<&'a [u8]> for ExtendedReport {
+    type Error = &'static str;
+
+    fn try_from(buf: &'a [u8]) -> Result<Self, Self::Error> {
+        if buf.len() < 4 {
+            return Err("Less than 4 bytes for ExtendedReport");
+        }
+
+        let sender_ssrc = u32::from_be_bytes(buf[..4].try_into().unwrap()).into();
+
+        let mut blocks: Vec<ReportBlock> = Vec::new();
+        let mut buf = &buf[4..];
+
+        while let Ok(block) = buf.try_into() {
+            let block: ReportBlock = block;
+            let len = block.len();
+            blocks.push(block);
+            buf = &buf[len..];
+        }
+
+        Ok(ExtendedReport {
+            sender_ssrc,
+            blocks,
+        })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ReportBlock {
+    ReceiverReferenceTime(ReceiverReferenceTime),
```

Comment:
Since we have Dlrr, let's call this Rrtr.

Google does it: https://webrtc.googlesource.com/src//+/d090952628606228afa47c842812927f15227290/modules/rtp_rtcp/source/rtcp_transceiver_config.h#172